src/awkward/_typetracer.py

```diff
@@ -935,102 +988,119 @@ def concatenate(self, arrays, casting="same_kind"):
             numpy.concatenate(emptyarrays).dtype, (UnknownLength,) + inner_shape
         )
 
-    def repeat(self, *args, **kwargs):
-        for x in args:
-            try_touch_data(x)
-        # array, int
-        # array1, array2
-        raise ak._errors.wrap_error(NotImplementedError)
-
-    def tile(self, *args, **kwargs):
-        for x in args:
-            try_touch_data(x)
-        # array, int
+    def repeat(
+        self,
+        x: TypeTracerArray,
+        repeats: TypeTracerArray | int,
+        *,
+        axis: int | None = None,
+    ) -> TypeTracerArray:
+        try_touch_data(x)
+        try_touch_data(repeats)
         raise ak._errors.wrap_error(NotImplementedError)
 
-    def stack(self, *args, **kwargs):
-        for x in args:
-            try_touch_data(x)
-        # arrays
+    def tile(self, x: TypeTracerArray, reps: int) -> TypeTracerArray:
+        try_touch_data(x)
         raise ak._errors.wrap_error(NotImplementedError)
 
-    def packbits(self, *args, **kwargs):
-        for x in args:
+    def stack(
+        self,
+        arrays: list[TypeTracerArray] | tuple[TypeTracerArray, ...],
+        *,
+        axis: int = 0,
+    ) -> TypeTracerArray:
+        for x in arrays:
             try_touch_data(x)
-        # array
         raise ak._errors.wrap_error(NotImplementedError)
 
-    def unpackbits(self, *args, **kwargs):
-        for x in args:
-            try_touch_data(x)
-        # array
+    def packbits(
+        self, array: TypeTracerArray, *, axis=None, bitorder="big"
+    ) -> TypeTracerArray:
```

Comment:
PyLint error. (I'm calling it out because they can be hard to find.)

Reply:
> How is maybe_reuse different from out? I see that it would no longer trouble TypeTracerArray because the TypeTracer functions simulate the situation by ignoring that argument. (It would be a side-effect whose success/failure status can only be determined at runtime.) This looks like a good implementation of out, rather than an alternative that needs a new name.

Precisely that. The hint is `maybe_` which means you cannot rely on it as a side-effect. I don't mind changing names from NumPy; this API layer is already doing that, and the Array API doesn't define anything for reusing arrays like this. 

Would you prefer `maybe_out`?

Reply:
Does this mean that we must always use it as

```python
outarray = nplike.function(inarray1, inarray2, maybe_out=outarray)
```

because if we did this:

```python
nplike.function(inarray1, inarray2, maybe_out=outarray)
```

then `outarray` would only get filled under some circumstances and not others?

(If that's the case, then definitely it needs a new name. `maybe_out` is nice because it links this with the already-known behavior of `out`.)

Reply:
Yes, the idea is that you *must* assume that a copy occurs each time from the perspective of using the result.

Fab, let's use `maybe_out`.